ethereum/core/src/main/java/org/hyperledger/besu/ethereum/mainnet/MainnetProtocolSpecs.java

```diff
@@ -1114,63 +1109,84 @@ static ProtocolSpecBuilder experimentalEipsDefinition(
         .hardforkId(EXPERIMENTAL_EIPS);
   }
 
-  private static TransactionReceipt frontierTransactionReceiptFactory(
-      // ignored because it's always FRONTIER
-      final TransactionType __,
-      final TransactionProcessingResult result,
-      final WorldState worldState,
-      final long gasUsed) {
-    return new TransactionReceipt(
-        worldState.frontierRootHash(),
-        gasUsed,
-        result.getLogs(),
-        Optional.empty()); // No revert reason in frontier
-  }
+  private static class FrontierTransactionReceiptFactory implements TransactionReceiptFactory {
 
-  private static TransactionReceipt byzantiumTransactionReceiptFactory(
-      // ignored because it's always FRONTIER
-      final TransactionType __,
-      final TransactionProcessingResult result,
-      final WorldState worldState,
-      final long gasUsed) {
-    return new TransactionReceipt(
-        result.isSuccessful() ? 1 : 0, gasUsed, result.getLogs(), Optional.empty());
+    @Override
+    public TransactionReceipt create(
+        final TransactionType transactionType,
+        final TransactionProcessingResult result,
+        final WorldState worldState,
+        final long gasUsed) {
+      return new TransactionReceipt(
+          worldState.frontierRootHash(),
+          gasUsed,
+          result.getLogs(),
+          Optional.empty()); // No revert reason in Frontier
+    }
+
+    @Override
+    public TransactionReceipt create(
+        final TransactionType transactionType,
+        final TransactionProcessingResult result,
+        final long gasUsed) {
+      throw new UnsupportedOperationException("No stateless transaction receipt in Frontier");
+    }
   }
 
-  private static TransactionReceipt byzantiumTransactionReceiptFactoryWithReasonEnabled(
-      // ignored because it's always FRONTIER
-      final TransactionType __,
-      final TransactionProcessingResult result,
-      final WorldState worldState,
-      final long gasUsed) {
-    return new TransactionReceipt(
-        result.isSuccessful() ? 1 : 0, gasUsed, result.getLogs(), result.getRevertReason());
+  private abstract static class StatelessTransactionReceiptFactory
```

Comment:
I'm not sure about the stateless naming, I think it can be missunderstood

Reply:
some alternatives:
1. `NoStateRequiredTransactionReceiptFactory`
2. `PostFrontierTransactionReceiptFactory`

Reply:
maybe that PostFrontierTransactionReceiptFactory